### ISSUE TITLE
Finetune Yoloe with coco128

### ISSUE THREAD

**Issue by tswc** [ultralytics/ultralytics#21833]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Following yoloe training docs to train coco128. 
It runs normally. But the mAP on training dataset is almost 0.
```python
from ultralytics import YOLOE
from ultralytics.models.yolo.yoloe import YOLOEPETrainer

# Initialize a detection model from a config
model = YOLOE("ultralytics-main/ultralytics/cfg/models/11/yoloe-11.yaml")

# Load weights from a pretrained segmentation checkpoint (same scale)
model.load("yoloe-11l-seg.pt")

# Fine-tune on your detection dataset
results = model.train(
    data="ultralytics/cfg/datasets/coco128.yaml",  # Detection dataset
    epochs=80,
    patience=10,
    trainer=YOLOEPETrainer,  # <- Important: use detection trainer
)
```

```
YOLOe-11 summary (fused): 104 layers, 2,616,248 parameters, 0 gradients, 6.5 GFLOPs
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 4/4 [00:02<00:00,  1.83it/s]
                   all        128        929   0.000951     0.0117   0.000482    0.00038
                person         61        254     0.0667    0.00394     0.0335     0.0268
               bicycle          3          6          0          0          0          0
                   car         12         46          0          0          0          0
            motorcycle          4          5          0          0          0          0
              airplane          5          6          0          0          0          0
                   bus          5          7          0          0          0          0
                 train          3          3   6.87e-05      0.333   5.06e-05   1.01e-05
                 truck          5         12          0          0          0          0
                  boat          2          6          0          0          0          0
         traffic light          4         14          0          0          0          0
             stop sign          2          2          0          0          0          0
                 bench          5          9          0          0          0          0
                  bird          2         16          0          0          0          0
                   cat          4          4          0          0          0          0
                   dog          9          9          0          0          0          0
                 horse          1          2          0          0          0          0
              elephant          4         17          0          0          0          0
                  bear          1          1          0          0          0          0
                 zebra          2          4          0          0          0          0
               giraffe          4          9   0.000275      0.111    0.00019   3.79e-05
              backpack          4          6          0          0          0          0
              umbrella          4         18          0          0          0          0
               handbag          9         19          0          0          0          0
                   tie          6          7          0          0          0          0
              suitcase          2          4          0          0          0          0
               frisbee          5          5          0          0          0          0
                  skis          1          1          0          0          0          0
             snowboard          2          7          0          0          0          0
           sports ball          6          6          0          0          0          0
                  kite          2         10          0          0          0          0
          baseball bat          4          4          0          0          0          0
        baseball glove          4          7          0          0          0          0
            skateboard          3          5          0          0          0          0
         tennis racket          5          7          0          0          0          0
                bottle          6         18          0          0          0          0
            wine glass          5         16          0          0          0          0
                   cup         10         36          0          0          0          0
                  fork          6          6          0          0          0          0
                 knife          7         16          0          0          0          0
                 spoon          5         22          0          0          0          0
                  bowl          9         28          0          0          0          0
                banana          1          1          0          0          0          0
              sandwich          2          2          0          0          0          0
                orange          1          4          0          0          0          0
              broccoli          4         11          0          0          0          0
                carrot          3         24          0          0          0          0
               hot dog          1          2          0          0          0          0
                 pizza          5          5          0          0          0          0
                 donut          2         14          0          0          0          0
                  cake          4          4          0          0          0          0
                 chair          9         35          0          0          0          0
                 couch          5          6          0          0          0          0
          potted plant          9         14          0          0          0          0
                   bed          3          3          0          0          0          0
          dining table         10         13   0.000533      0.385   0.000485   0.000155
                toilet          2          2          0          0          0          0
                    tv          2          2          0          0          0          0
                laptop          2          3          0          0          0          0
                 mouse          2          2          0          0          0          0
                remote          5          8          0          0          0          0
            cell phone          5          8          0          0          0          0
             microwave          3          3          0          0          0          0
                  oven          5          5          0          0          0          0
                  sink          4          6          0          0          0          0
          refrigerator          5          5          0          0          0          0
                  book          6         29          0          0          0          0
                 clock          8          9          0          0          0          0
                  vase          2          2          0          0          0          0
              scissors          1          1          0          0          0          0
            teddy bear          6         21          0          0          0          0
            toothbrush          2          5          0          0          0          0
```

**Comment by UltralyticsAssistant** [comment]
👋 Hello @tswc, thank you for your interest in Ultralytics 🚀! This is an automated triage to help move things forward quickly. An Ultralytics engineer will also assist you here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. From your description (very low mAP on COCO128 while fine-tuning), please include:
- Exact script or notebook cell(s) used (including model config and weights paths)
- Full training logs from start to finish (first few and last few epochs)
- Your environment details: OS, Python, PyTorch, CUDA versions, and GPU model
- Ultralytics package version and how it was installed
- The exact dataset YAML used and confirmation images/labels load correctly
- Any custom modifications (trainer, loss, hyperparameters, or config edits)

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Additional note for your case:
- Since you’re fine-tuning with a detection trainer, please confirm the pretrained weights you load are compatible with detection heads for your model scale and that the model’s nc in the YAML matches your dataset classes. Sharing these specifics in your MRE will help us reproduce quickly ✅

**Comment by RizwanMunawar** [comment]
@tswc hi, I just performed testing and everything is working well. I am not able to reproduce it. Thanks

**Code**
```python
from ultralytics import YOLOE
from ultralytics.models.yolo.yoloe import YOLOEPETrainer

# Initialize a detection model from a config
model = YOLOE("yoloe-11s.yaml")

# Load weights from a pretrained segmentation checkpoint (same scale)
model.load("yoloe-11s-seg.pt")

# Fine-tune on your detection dataset
results = model.train(
    data="coco128.yaml",  # Detection dataset
    epochs=5,
    patience=10,
    trainer=YOLOEPETrainer,  # <- Important: use detection trainer
)
```

**Logs**
```
YOLOe-11s summary (fused): 104 layers, 9,443,760 parameters, 0 gradients, 21.5 GFLOPs
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 4/4 1.8it/s 2.2s
                   all        128        929      0.571      0.507       0.54      0.397
                person         61        254      0.755      0.669      0.755      0.526
               bicycle          3          6      0.333      0.333      0.434      0.226
                   car         12         46      0.504      0.217      0.243      0.138
            motorcycle          4          5      0.668        0.8      0.866      0.682
              airplane          5          6      0.622      0.833      0.821      0.747
                   bus          5          7      0.512      0.571      0.649      0.583
                 train          3          3      0.696          1      0.913       0.78
                 truck          5         12      0.681       0.25      0.464      0.277
                  boat          2          6      0.309      0.167      0.301      0.196
         traffic light          4         14      0.496      0.214      0.286      0.173
             stop sign          2          2       0.57        0.5      0.512      0.458
                 bench          5          9       0.54      0.395       0.51       0.33
                  bird          2         16      0.756      0.438      0.808      0.471
                   cat          4          4        0.5          1      0.995      0.828
                   dog          9          9      0.553          1      0.898      0.709
                 horse          1          2      0.248          1      0.995      0.746
              elephant          4         17      0.858      0.824      0.868      0.744
                  bear          1          1      0.575          1      0.995      0.995
                 zebra          2          4       0.84          1      0.995      0.946
               giraffe          4          9      0.725          1      0.963      0.703
              backpack          4          6      0.593      0.333      0.323      0.186
              umbrella          4         18      0.751      0.671      0.733      0.384
               handbag          9         19      0.398      0.368        0.3      0.166
                   tie          6          7      0.661      0.571      0.576       0.37
              suitcase          2          4      0.744       0.25      0.515      0.395
               frisbee          5          5      0.757        0.6      0.716      0.627
                  skis          1          1      0.204          1      0.497      0.398
             snowboard          2          7      0.469      0.384      0.517      0.363
           sports ball          6          6      0.258      0.167      0.234      0.127
                  kite          2         10      0.717        0.2      0.247     0.0936
          baseball bat          4          4      0.582        0.5       0.39      0.197
        baseball glove          4          7       0.54      0.429      0.505      0.253
            skateboard          3          5      0.367        0.2      0.358      0.167
         tennis racket          5          7      0.797      0.429      0.491      0.262
                bottle          6         18      0.458      0.611      0.375      0.218
            wine glass          5         16      0.623      0.438      0.441      0.244
                   cup         10         36      0.464      0.417       0.41        0.3
                  fork          6          6      0.194      0.167      0.188      0.177
                 knife          7         16      0.457      0.375      0.378      0.278
                 spoon          5         22      0.435      0.246       0.27      0.152
                  bowl          9         28      0.658      0.688      0.587      0.489
                banana          1          1          1          0     0.0276    0.00276
              sandwich          2          2          0          0      0.165      0.165
                orange          1          4          1          0      0.639      0.432
              broccoli          4         11      0.586      0.182      0.171      0.151
                carrot          3         24          1     0.0719      0.612      0.362
               hot dog          1          2      0.444          1      0.828      0.828
                 pizza          5          5      0.595          1      0.895      0.785
                 donut          2         14      0.679      0.857       0.86      0.748
                  cake          4          4      0.246          1      0.745      0.679
                 chair          9         35      0.358      0.571       0.39      0.241
                 couch          5          6      0.395      0.667      0.751      0.588
          potted plant          9         14      0.717      0.544      0.621      0.319
                   bed          3          3      0.797          1      0.995      0.548
          dining table         10         13      0.602      0.615      0.598      0.318
                toilet          2          2      0.604        0.5      0.527      0.469
                    tv          2          2      0.103        0.5      0.586        0.5
                laptop          2          3          1          0      0.131      0.109
                 mouse          2          2          1          0      0.197     0.0492
                remote          5          8      0.552      0.375      0.563      0.459
            cell phone          5          8      0.428       0.25      0.171      0.121
             microwave          3          3      0.387      0.667      0.806      0.441
                  oven          5          5     0.0985        0.2      0.145     0.0465
                  sink          4          6      0.379       0.31      0.234      0.172
          refrigerator          5          5      0.601        0.8      0.819      0.615
                  book          6         29      0.468      0.172      0.206      0.118
                 clock          8          9      0.833      0.889      0.888      0.668
                  vase          2          2      0.208          1      0.606      0.551
              scissors          1          1          1          0          0          0
            teddy bear          6         21      0.616      0.571      0.599      0.453
            toothbrush          2          5          1          0      0.245      0.146
```

**Comment by tswc** [comment]
```
> [@tswc](https://github.com/tswc) hi, I just performed testing and everything is working well. I am not able to reproduce it. Thanks
> 
> **Code**
> 
> from ultralytics import YOLOE
> from ultralytics.models.yolo.yoloe import YOLOEPETrainer
> 
> # Initialize a detection model from a config
> model = YOLOE("yoloe-11s.yaml")
> 
> # Load weights from a pretrained segmentation checkpoint (same scale)
> model.load("yoloe-11s-seg.pt")
> 
> # Fine-tune on your detection dataset
> results = model.train(
>     data="coco128.yaml",  # Detection dataset
>     epochs=5,
>     patience=10,
>     trainer=YOLOEPETrainer,  # <- Important: use detection trainer
> )
```

Thanks.
where is yoloe-11s.yaml? I haven't found it in config folder

**Comment by tswc** [comment]
all my god. I got it.

**Comment by glenn-jocher** [comment]
Nice—glad you found it. One tip: ensure the model scale in your YAML matches the pretrained segmentation checkpoint you load (e.g., `yoloe-11s.yaml` with `yoloe-11s-seg.pt`); mixing scales can lead to near‑zero mAP. If results are still off, please share `yolo checks` output and the full training log for review. See the [YOLOE fine-tuning guide](https://docs.ultralytics.com/models/yoloe/) and the [COCO128 dataset page](https://docs.ultralytics.com/datasets/detect/coco128/) for reference.
